Backpack/Dialog/Classes/BPKDialogView.m

```diff
@@ -53,21 +54,23 @@ @implementation BPKDialogView
 
 - (instancetype)initWithTitle:(NSString *)title
                       message:(NSString *)message
-          iconBackgroundColor:(UIColor *)iconBackgroundColor
-                    iconImage:(UIImage *)iconImage {
+               iconDefinition:(BPKDialogIconDefinition *_Nullable)iconDefinition {
     BPKAssertMainThread();
     self = [self initWithFrame:CGRectZero];
 
     if (self) {
         self.title = title;
         self.message = message;
-        self.iconBackgroundColor = iconBackgroundColor;
-        self.iconImage = iconImage;
+        self.iconDefinition = iconDefinition;
     }
 
     return self;
 }
 
+- (Boolean)hasIcon {
```

Comment:
BOOL

Reply:
Shouldn't this be delcared as a readonly property too?

Reply:
Tempted to setup a danger rule for this as it always catches me out!